clickhouse/client.cpp

```diff
@@ -167,6 +190,7 @@ class Client::Impl {
 #endif
 
     ServerInfo server_info_;
+    std::optional<ClientOptions::HostPort> connected_host_port_;
 };
 
 
```

Comment:
Please modify endpoints creation to simplify the `ResetConnection()`
```suggestion
ClientOptions modifyClientOptions(ClientOptions opts)
{
    if (!opts.host.empty())
        opts.endpoints.insert(opts.endpoints.begin(), ClientOptions::Endpoint{opts.host, opts.port});

    return opts;
}

Client::Impl::Impl(const ClientOptions& opts)
    : options_(modifyClientOptions(opts))
{
```